runtime/compiler/env/J9ClassEnv.hpp

```diff
@@ -121,6 +121,21 @@ class OMR_EXTENSIBLE ClassEnv : public OMR::ClassEnvConnector
     */
    bool classHasIdentity(TR_OpaqueClassBlock *clazz);
 
+   /** \brief
+    *    Checks whether a class contains any field of type double (D), float (F),
+    *    nullable-class/interface type (L) or null-free class type (Q) that are
+    *    not both flattened and recursively compatible for the fast substitutability
+    *    optimization.
```

Comment:
What this checks for might be a little unclear from this description.  I would suggest saying that it checks whether the class is compatible with the fast substitutability optimization (or if you use a more meaningful name, whatever you settle on), followed by a list of what it checks for:

> that no field is of type double (D) or float (F)
> etc.

Reply:
Addressed in 307194d